src/inferencer/inferencer.ts

```diff
@@ -1,37 +1,46 @@
 import { ancestor } from 'acorn-walk/dist/walk'
-import { TypeAnnotatedNode } from '../types'
+import { TypeAnnotatedNode, Variable } from '../types'
 import { annotateProgram } from './annotator'
+import { primitiveMap } from './typeEnvironment'
 import * as es from 'estree'
 
 // // main function that will infer a program
 export function inferProgram(program: es.Program): TypeAnnotatedNode<es.Program> {
+  const typeConstraints = new Map()
```

Comment:
Might be a good idea to have another file for type constraint related logic, so we don't clutter this file.